### PR TITLE
Test authority system

### PR DESCRIPTION
**Pull request by k-gregory**

### CODE REVIEW

**Comment by kostya9** [comment]
@k-gregory Could you please summarize the changes in a brief message?

**Comment by k-gregory** [comment]
@kostya9 
This PR configures integration testing. It uses embedded H2 databases for it and automatic schema generation.
Also it adds tests for user permissions system. I guess this test shall be splitted, but I am not sure.

**Comment by kostya9** [comment]
@k-gregory Could we use a real postgresql database that Travis can provide?
https://docs.travis-ci.com/user/database-setup/#PostgreSQL

Or we will support H2 migration scripts?

**Comment by k-gregory** [comment]
@kostya9 We won't support H2 migration scripts. As long as Hibernate successfully creates schema and we don't use native PostgreSQL features, adding PostgreSQL can be done in separate PR

**Review by kostya9 — APPROVED** [review]

File: backend/src/test/java/io/github/k_gregory/registry/service/SpringDataUserServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Autowired
+    private SpringDataUserService userService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Test
+    @Transactional
+    public void getUserAuthorities() {
```
Comment (kostya9) [inline backend/src/test/java/io/github/k_gregory/registry/service/SpringDataUserServiceTest.java]: Could we be more specific about the purpose of this test? We should not fear that the test names will be long - that's a good practice.
Let's change the name to mirror the purpose.

**Review by kostya9 — CHANGES_REQUESTED** [review]

File: backend/src/test/java/io/github/k_gregory/registry/service/SpringDataUserServiceTest.java (hunk 1 above)
Comment (k-gregory) [reply backend/src/test/java/io/github/k_gregory/registry/service/SpringDataUserServiceTest.java]: It tests method named 'getUserAuthorities' in 'SpringDataUserService' class. This method takes `AppUser`, fetches all user permission from DB and all permissions of groups to which user belongs. In this test I create 4 permissions, 3 groups and 4 users. Then I assign some rights to these users and add them to some groups. Then I compare set of permissions returned by `getUserAuthorities` to set of permissions which is correct. 
Which name should I use and would you suggest me to split this method?

Comment (kostya9) [reply backend/src/test/java/io/github/k_gregory/registry/service/SpringDataUserServiceTest.java]: @k-gregory I would suggest naming that something like
public void getUserAuthorities_WhenAddingAuthoritiesToUsersAndGroups_ReturnsAllAuthorities

That's if I understood the case correctly

**Review by kostya9 — APPROVED** [review]

**Pull request merged**